src/main/java/io/stargate/sgv2/jsonapi/service/operation/MetadataAttemptPage.java

```diff
@@ -0,0 +1,74 @@
+package io.stargate.sgv2.jsonapi.service.operation;
+
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResultBuilder;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.KeyspaceSchemaObject;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaObject;
+import java.util.function.Supplier;
+
+/**
+ * A page of results for metadata based command, use {@link #builder()} to get a builder to pass to
+ * {@link GenericOperation}.
+ */
+public abstract class MetadataAttemptPage<SchemaT extends SchemaObject>
+    extends OperationAttemptPage<SchemaT, MetadataAttempt<SchemaT>> {
+
+  private final boolean showSchema;
+  private final CommandStatus statusKey;
+
+  private MetadataAttemptPage(
+      OperationAttemptContainer<SchemaT, MetadataAttempt<SchemaT>> attempts,
+      CommandResultBuilder resultBuilder,
+      boolean showSchema,
+      CommandStatus statusKey) {
+    super(attempts, resultBuilder);
+    this.showSchema = showSchema;
+    this.statusKey = statusKey;
+  }
+
+  public static <SchemaT extends KeyspaceSchemaObject>
+      MetadataAttemptPage.Builder<SchemaT> builder() {
+    return new MetadataAttemptPage.Builder<>();
+  }
+
+  @Override
+  protected void buildCommandResult() {
+    addAttemptWarningsToResult();
+    addAttemptErrorsToResult();
+    if (attempts.get(0).status() != OperationAttempt.OperationStatus.ERROR) {
+      if (showSchema) {
+        resultBuilder.addStatus(statusKey, attempts.get(0).getSchema());
+      } else {
+        resultBuilder.addStatus(statusKey, attempts.get(0).getNames());
+      }
+    }
+  }
+
+  public static class Builder<SchemaT extends SchemaObject>
+      extends OperationAttemptPageBuilder<SchemaT, MetadataAttempt<SchemaT>> {
+
+    private boolean showSchema = false;
+    private CommandStatus statusKey;
+
+    Builder() {}
+
+    public Builder<SchemaT> showSchema(boolean showSchema) {
+      this.showSchema = showSchema;
+      return this;
+    }
+
+    public Builder<SchemaT> usingCommandStatus(CommandStatus statusKey) {
+      this.statusKey = statusKey;
+      return this;
+    }
+
+    public Supplier<CommandResult> getOperationPage() {
+      var resultBuilder =
+          new CommandResultBuilder(
```

Comment:
I changed the way this was build in this PR, still not merged cause of failing IT https://github.com/stargate/data-api/pull/1519

leave this for now it can change when this is done https://github.com/stargate/data-api/issues/1518